src/test/groovy/org/prebid/server/functional/tests/CacheSpec.groovy

```diff
@@ -93,4 +94,100 @@ class CacheSpec extends BaseSpec {
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            enableCache()
+            ext.prebid.cache.bids.returnCreative = false
+        }
+
+        and: "Default basic bid with banner creative"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid[0].bid[0].adm = new Adm(assets: [new Asset(id: PBSUtils.randomNumber)])
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response shouldn't contain adm obj"
+        assert !response.seatbid[0].bid[0].adm
+    }
+
+    def "PBS should response with seatbid.bid.adm in response when ext.prebid.cache.bids.returnCreative=true"() {
+        given: "Default BidRequest with cache, targeting"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            enableCache()
+            ext.prebid.cache.bids.returnCreative = true
+        }
+
+        and: "Default basic bid with banner creative"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid[0].bid[0].adm = new Adm(assets: [new Asset(id: PBSUtils.randomNumber)])
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain adm obj"
+        assert response.seatbid[0].bid[0].adm == bidResponse.seatbid[0].bid[0].adm
+    }
+
+    def "PBS shouldn't response with seatbid.bid.adm in response when ext.prebid.cache.vastXml.returnCreative=false and video request"() {
+        given: "Default BidRequest with cache, targeting"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            imp[0].banner = null
+            imp[0].video = Video.defaultVideo
+            enableCache()
+            ext.prebid.cache.vastXml.returnCreative = false
+        }
+
+        and: "Default basic bid with banner creative"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid[0].bid[0].adm = new Adm(assets: [new Asset(id: PBSUtils.randomNumber)])
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response shouldn't contain adm obj"
+        assert !response.seatbid[0].bid[0].adm
+    }
+
+    def "PBS should response with seatbid.bid.adm in response when ext.prebid.cache.vastXml.returnCreative=#returnCreative and #request"() {
+        given: "Default BidRequest with cache, targeting"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            enableCache()
+            imp[0].video = video
```

Comment:
In this case, we will have a video and banner for imp. Maybe we can use `imp[0] = Imp.getDefaultImpression(mediaType)` and change `video` to `mediaType` and remove `request`